django/db/models/base.py

```diff
@@ -569,6 +569,10 @@ def _get_pk_val(self, meta=None):
         return getattr(self, meta.pk.attname)
 
     def _set_pk_val(self, value):
+        for parent_link in self._meta.parents.values():
+            if parent_link:
+                setattr(self, parent_link.attname, value)
+                setattr(self, parent_link.target_field.attname, value)
```

Comment:
Is this assignment redundant with the above one given we walk over all parents?

Reply:
First sets pk for a main model, second for parents. For example, in added test we have
- `parent_link`:

  ```
  model_inheritance_regress.Congressman.person_ptr_id
  model_inheritance_regress.Congressman.politician_ptr_id
  ```

- `parent_link.target_field`:

  ```
  model_inheritance_regress.Person.id
  model_inheritance_regress.Politician.politician_id
  ```

We can remove the first one and force users to set an "ID" explicitly (TBH it's even documented that way :thinking: ), e.g.
```
c2.pk = None
c2.id = None
c2.politician_ptr_id = None
```

Reply:
yeah we should stick to setting `pk`s.

Reply:
Agreed, I also skip a parent link if it was promoted to a primary key.